sklearn/ensemble/_gb.py

```diff
@@ -203,27 +418,39 @@ def _fit_stage(
         X_csc=None,
         X_csr=None,
     ):
-        """Fit another stage of ``_n_classes`` trees to the boosting model."""
-
-        assert sample_mask.dtype == bool
-        loss = self._loss
+        """Fit another stage of ``n_trees_per_iteration_`` trees."""
+        if sample_mask.dtype != bool:
+            raise ValueError("Only dtype bool is allowed for sample_mask.")
```

Comment:
This is never called because `_fit_stages` always constructions `sample_mask` properly, thus codecov is complaining.

I am okay with leaving the `assert` or removing the check all together.

Reply:
I'll remove it.